pallets/identity/src/lib.rs

```diff
@@ -1766,7 +1792,11 @@ impl<T: Trait> Module<T> {
     fn unlink_key_to_did(key: &AccountKey, did: IdentityId) {
         if let Some(linked_key_info) = <KeyToIdentityIds>::get(key) {
             match linked_key_info {
-                LinkedKeyInfo::Unique(..) => <KeyToIdentityIds>::remove(key),
+                LinkedKeyInfo::Unique(did_linked) => {
+                    if did_linked == did {
```

Comment:
I think we can ignore this if as in `leave_identity()` you are already checking valid signer status

Reply:
`leave_identity` has this check but not some other functions.